### PR TITLE
Improve error message when git ref cannot be fetched

### PR DESCRIPTION
**Pull request by zanieb**
Follow-up to #1781 improving the error message when a ref cannot be fetched

### CODE REVIEW

File: crates/uv/tests/common/mod.rs [hunk 1]
```diff
@@ -148,12 +148,14 @@ impl TestContext {
         ]
     }
 
-    /// Canonical snapshot filters for this test context.
+    /// Standard snapshot filters _plus_ those for this test context.
     pub fn filters(&self) -> Vec<(&str, &str)> {
```
Comment (snowsignal) [inline crates/uv/tests/common/mod.rs:152]: We could simplify this function to be:
```rust
    pub fn filters(&self) -> Vec<(&str, &str)> {
        // Put test context snapshots before the default filters
        // This ensures we don't replace other patterns inside paths from the test context first
        self.filters
            .iter()
            .chain(INSTA_FILTERS.into_iter())
            .copied()
            .collect()
    }
```
which lets us avoid the repeated `insert` calls.

**Review by snowsignal — APPROVED** [review]

File: crates/uv/tests/common/mod.rs (hunk 1 above)
Comment (zanieb) [reply crates/uv/tests/common/mod.rs:152]: That fails

```
error[E0271]: type mismatch resolving `<Iter<'_, (&str, &str)> as IntoIterator>::Item == &(String, String)`
   --> crates/uv/tests/common/mod.rs:157:20
    |
157 |             .chain(INSTA_FILTERS.into_iter())
    |              ----- ^^^^^^^^^^^^^^^^^^^^^^^^^ expected `&(String, ...)`, found `&(&str, &str)`
    |              |
    |              required by a bound introduced by this call
    |
    = note: expected reference `&(std::string::String, std::string::String)`
               found reference `&(&str, &str)`
note: the method call chain might not have had the expected associated types
   --> crates/uv/tests/common/mod.rs:157:34
    |
157 |             .chain(INSTA_FILTERS.into_iter())
    |                    ------------- ^^^^^^^^^^^ `IntoIterator::Item` is `&(&str, &str)` here
    |                    |
    |                    this expression has type `&[(&str, &str)]`
note: required by a bound in `std::iter::Iterator::chain`
   --> /Users/mz/.rustup/toolchains/1.76-aarch64-apple-darwin/lib/rustlib/src/rust/library/core/src/iter/traits/iterator.rs:524:25
    |
521 |     fn chain<U>(self, other: U) -> Chain<Self, U::IntoIter>
    |        ----- required by a bound in this associated function
...
524 |         U: IntoIterator<Item = Self::Item>,
    |                         ^^^^^^^^^^^^^^^^^ required by this bound in `Iterator::chain`

error[E0599]: the method `copied` exists for struct `Chain<Iter<'_, (String, ...)>, ...>`, but its trait bounds were not satisfied
   --> crates/uv/tests/common/mod.rs:158:14
    |
155 | /         self.filters
156 | |             .iter()
157 | |             .chain(INSTA_FILTERS.into_iter())
158 | |             .copied()
    | |             -^^^^^^ method cannot be called on `Chain<Iter<'_, (String, ...)>, ...>` due to unsatisfied trait bounds
    | |_____________|
    | 
 ```

Comment (snowsignal) [reply crates/uv/tests/common/mod.rs:152]: Okay, what about this?
```rust
    pub fn filters(&self) -> Vec<(&str, &str)> {
        // Put test context snapshots before the default filters
        // This ensures we don't replace other patterns inside paths from the test context first
        self.filters
            .iter()
            .map(|p, r| (p.as_str(), r.as_str()))
            .chain(INSTA_FILTERS.into_iter().copied())
            .collect()
    }
```

Comment (zanieb) [reply crates/uv/tests/common/mod.rs:152]: I'm not sure it's any easier to read but it does work!

Comment (snowsignal) [reply crates/uv/tests/common/mod.rs:152]: It's easier to read for me, but maybe I'm just accustomed to using iterators for everything 😅 

If you want an approach that avoids iterator chaining, what about this?
```rust
    pub fn filters(&self) -> Vec<(&str, &str)> {
        let mut filters: Vec<(&str, &str)> = vec![];
        // Put test context snapshots before the default filters
        // This ensures we don't replace other patterns inside paths from the test context first
        for (pattern, replacement) in &self.filters {
            filters.push((pattern, replacement));
        }
        filters.extend(INSTA_FILTERS.iter());
        filters
    }
```

**Pull request merged**